### ISSUE TITLE
RDS Integration for Cloudwatch Logs MySQL Handler fails silently

### ISSUE THREAD

**Issue by fernandrone** [honeycombio/agentless-integrations-for-aws#28]
Hey :wave: 

I noticed this when I invalidated my API key by accident, the logs stopped coming in, but the Lambda function was succeeding. There wasn't even any error logs.

Looking at the code, the problem does not seem limited to authentication errors; here is the API call to send events on the MySQL Handler: 

https://github.com/honeycombio/agentless-integrations-for-aws/blob/master/mysql-handler/main.go#L84

This function return an error, but it is ignored: https://github.com/honeycombio/agentless-integrations-for-aws/blob/1744fea017d9bfe480ca42506423885e5df91c62/vendor/github.com/honeycombio/libhoney-go/libhoney.go#L750

The error should certainly be logged. I guess it's arguable if the lambda function should:

a. fail if any error happens
b. succeed independently of the number of errors (these are parallel goroutines)
c. fail if a certain % of goroutines fail?

Also, I _think_ (but I didn't look at it) that the PostgreSQL handler has the same issue, because when testing it I remember that it didn't log anything and the calls were succeeding, although the parsing was failing (later I learned from the Honeycomb team that the Agentless integration for PostgreSQL is deprecated and the `rdslogs` binary should be used instead).

**Comment by mollystamos123** [comment]
I have another instance of this: https://app.intercom.com/a/apps/u7i8awj8/inbox/inbox/all/conversations/27214224456

**Comment by tredman** [comment]
FWIW, the PostgreSQL agent is not deprecated. We did have some language to that effect in the past (if you still see this somewhere let me know and we'll fix  that), but the general rule is:
- `rdslogs` is the easiest / cheapest way to go for most use cases. It has trouble with very large log volumes, though, since it's blocked on the throughput of the RDS API. At a larger log volumes, Lambda is the only way to go to reliably get the RDS logs data.
- Lambda / Cloudwatch is also supported, but beware the ingest costs for Cloudwatch (they charge $.50/GB for ingest).

I'll take a look at the error logging behavior here. I think we definitely should return an error so that Lambda counts it as a "failed" invocation. But excessive error logging is dangerous due to the high ingest costs in Cloudwatch. In general, we've tried to keep the amount of logging to a minimum.

**Comment by JamieDanielson** [comment]
Hello,

As we enter 2022 Q1, we are trimming our OSS backlog. We will be closing this issue as it is a low priority for us to fix. It is unlikely that we'll ever get to it, and so we'd like to set expectations accordingly.

If this issue is important to you, please feel free to ping here and we can discuss/re-open.
